extensions/amp-instagram/0.1/amp-instagram.js

```diff
@@ -85,8 +86,10 @@ class AmpInstagram extends AMP.BaseElement {
     const image = this.win.document.createElement('amp-img');
     // This will redirect to the image URL. By experimentation this is
     // always the same URL that is actually used inside of the embed.
-    image.setAttribute('src', 'https://www.instagram.com/p/' +
-        encodeURIComponent(this.shortcode_) + '/media/?size=l');
+    const imgSrc = resolveUrlAttr('amp-img', 'src',
```

Comment:
There would actually be no extra latency. The current request pattern is

1. Request to instagram
2. Redirect to CDN.

I don't think it'll work, though. The API @erwinmombay mentioned is designed for JSONP (no CORS support), then their real API (not the oembed endpoint) requires an access token.

I think for now we can just guard the existing code behind `whenFirstVisible` promise. Having an instagram in the first viewport is quite uncommon anyway, which makes this essentially a null change.